src/live_restore/live_restore_fs.c

```diff
@@ -1194,46 +1194,59 @@ __live_restore_fs_open_in_source(WTI_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL *se
     return (ret);
 }
 
+/*
+ * __live_restore_compute_nbits --
+ *     Compute the number of bits needed for the bitmap, based off the destination file size.
+ */
+static int
+__live_restore_compute_nbits(
+  WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, uint64_t *nbitsp)
+{
+    wt_off_t size;
+    WT_RET(lr_fh->destination->fh_size(lr_fh->destination, (WT_SESSION *)session, &size));
+    WT_ASSERT_ALWAYS(session, size % lr_fh->allocsize == 0,
+      "The file size isn't a multiple of the file allocation size!");
+    *nbitsp = (uint64_t)size / (uint64_t)lr_fh->allocsize;
+    return (0);
+}
+
 /*
  * __live_restore_decode_bitmap --
  *     Decode a bitmap from a hex string.
  */
 static int
-__live_restore_decode_bitmap(WT_SESSION_IMPL *session, const char *bitmap_str, uint64_t nbits,
-  WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh)
+__live_restore_decode_bitmap(WT_SESSION_IMPL *session, const char *bitmap_str,
+  uint64_t metadata_nbits, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh)
 {
     WT_DECL_RET;
     WT_ASSERT_ALWAYS(session, bitmap_str != NULL, "Live restore bitmap string is NULL");
 
-    WT_RET(__bit_alloc(session, nbits, &lr_fh->bitmap));
-    lr_fh->nbits = nbits;
+    WT_RET(__bit_alloc(session, metadata_nbits, &lr_fh->bitmap));
+    lr_fh->nbits = metadata_nbits;
 
     WT_ITEM buf;
     WT_CLEAR(buf);
     WT_ERR(__wt_hex_to_raw(session, bitmap_str, &buf));
     memcpy(lr_fh->bitmap, buf.mem, buf.size);
-    /* FIXME-WT-13813: Add check here for shorter dest than bitmap, set end to 1. */
+
+    uint64_t file_size_nbits;
+    WT_ERR(__live_restore_compute_nbits(session, lr_fh, &file_size_nbits));
+    /*
+     * We may have truncated a file and the filesize on disk is shorter than that in the bitmap
+     * itself. This would happen even without a crash, we can assert that all the bits in the file
+     * past the file size nbits are set.
+     */
+    if (file_size_nbits < metadata_nbits)
+        for (uint64_t i = file_size_nbits; i < metadata_nbits; i++)
+            WT_ASSERT_ALWAYS(session, __bit_test(lr_fh->bitmap, i),
```

Comment:
If the bitmap can be incorrect after the truncate range, isn't it possible the bitmap is also incorrect before the truncate range? i.e. When we run into this scenario how can we be sure that there aren't false holes at the start of the file because we failed to write the metadata?

I think we should consider crashing in all cases where the file size doesn't match the bitmap size.

Reply:
File size not matching bitmap size is expected, because the destination file may grow or shrink so I am not sure what your suggestion means.

> When we run into this scenario how can we be sure that there aren't false holes at the start of the file because we failed to write the metadata?
I am somewhat confused by this, lets discuss on a call.

Reply:
Discussed this offline. The summary is that there may be issues in the rest of the bitmap, but we don't have a way to verify those bits. We can only verify the bits in the truncated range as we know they must be set.